garageScript/c0d3-app
# feat: Make `AdminLessonInputs` generic

Related to #2081 

### Description

This PR aims at making the [AdminLessonInputs](https://github.com/garageScript/c0d3-app/blob/master/components/admin/lessons/AdminLessonInputs/AdminLessonInputs.tsx) component generic by updating it to take the hooks for adding/updating the item as props and make its types generic.

### Small features

- Show a custom error message for empty inputs
- Change all references for `MODULE` with `ITEM`

### Concerns

I had trouble trying to make its types generic. I'm aware it's not perfect. Please let me know in the comments how it could be improved.

### Testing

1. Go to `/admin/lessons/js1/modules`
2. Fill in the inputs and hit `ADD ITEM`
3. Make sure it works
